packages/ui-motion/src/Transition/BaseTransition/index.ts

```diff
@@ -119,7 +119,9 @@ class BaseTransition extends React.Component<
     }
 
     if (prevProps.in !== this.props.in) {
-      this.startTransition(this.props.in, true)
+      if (_prevState.transitioning)
+        setTimeout(() => this.startTransition(this.props.in, true), 300)
```

Comment:
please try to solve this without `setTimeout`. e.g. what if this is called multiple times in a single frame? or its the delay is set to much more than 300ms? A proper way would be to cancel the ongoing transition and start the new one requested.